### PR TITLE
[IMP]Improved kanban m2m field, if special options like m2m_tag=True …

### PR DESCRIPTION
**Pull request by msh-odoo**
…passed from view then render many2many_tags else render color_bar, applied m2m_tag=True in project's and task's kanban view, by default keep color_bar if no special options are passed from view

Task: https://www.odoo.com/web#id=19439&view_type=form&model=project.task&action=327
Pad: https://pad.odoo.com/p/r.V9RDA56FKDBRQV8A

### CODE REVIEW

File: addons/web_kanban/static/src/js/kanban_view.js [hunk 1]
```diff
@@ -509,12 +509,16 @@ var KanbanView = View.extend({
             var dataset = new data.DataSetSearch(self, rel_name, self.dataset.get_context(rel.context));
             dataset.name_get(_.uniq(rel.ids)).done(function(result) {
                 result.forEach(function(nameget) {
-                    // white color (0) should not be selected, 1 instead
-                    var color = KanbanRecord.prototype.kanban_getcolor(nameget[1]) + 1;
-                    var $tag = $('<span>')
-                        .addClass('o_tag oe_kanban_color_' + color)
-                        .attr('title', _.str.escapeHTML(nameget[1]));
-                    $(rel.elements[nameget[0]]).append($tag);
+                    if ($(rel.elements[nameget[0]]).hasClass('o_m2m_tags')) {
```
Comment (hmo-odoo) [inline addons/web_kanban/static/src/js/kanban_view.js]: can not use variable to store $(rel.elements[nameget[0]]) ?

File: addons/web_kanban/static/src/js/kanban_view.js [hunk 2]
```diff
@@ -697,7 +702,16 @@ function transform_qweb_template (node, fvg, many2manys) {
...
+                var tag_class = ''
+                if (node.attrs.options) {
+                    var options = pyeval.py_eval(node.attrs.options);
+                    if (options.m2m_tags) {
+                        tag_class = ' o_m2m_tags';
+                    }
+                } else {
+                    tag_class = ' o_kanban_color_tags';
+                }
+                node.attrs['class'] = (node.attrs['class'] || '') + tag_class + ' oe_form_field oe_tags';
```
Comment (hmo-odoo) [inline addons/web_kanban/static/src/js/kanban_view.js]: can not use sprintf method of underscore to join string ?

File: addons/web_kanban/static/src/js/kanban_view.js [hunk 3]
```diff
@@ -697,7 +702,16 @@ function transform_qweb_template (node, fvg, many2manys) {
                     many2manys.push(node.attrs.name);
                 }
                 node.tag = 'div';
-                node.attrs['class'] = (node.attrs['class'] || '') + ' oe_form_field oe_tags o_kanban_tags';
+                var tag_class = ''
```
Comment (hmo-odoo) [inline addons/web_kanban/static/src/js/kanban_view.js]: ; missing

File: addons/web_kanban/static/src/js/kanban_view.js [hunk 4]
```diff
@@ -510,12 +510,17 @@ var KanbanView = View.extend({
...
             dataset.name_get(_.uniq(rel.ids)).done(function(result) {
                 result.forEach(function(nameget) {
-                    // white color (0) should not be selected, 1 instead
-                    var color = KanbanRecord.prototype.kanban_getcolor(nameget[1]) + 1;
-                    var $tag = $('<span>')
-                        .addClass('o_tag oe_kanban_color_' + color)
-                        .attr('title', _.str.escapeHTML(nameget[1]));
-                    $(rel.elements[nameget[0]]).append($tag);
+                    var $element = $(rel.elements[nameget[0]]);
+                    if ($element.hasClass('o_m2m_tags')) {
```
Comment (tde-banana-odoo) [inline addons/web_kanban/static/src/js/kanban_view.js]: Can't you use the options instead of relying on a class ?

Comment (msh-odoo) [inline addons/web_kanban/static/src/js/kanban_view.js]: Well in postprocess_m2m_tags method we will not have field options available directly, yes we can have options as attribute of $element, we can eval it using pyeval and use it, or we can create a m2m_options object, something like m2m_context for each m2m field

Comment (vme-odoo) [inline addons/web_kanban/static/src/js/kanban_view.js]: @msh-odoo we cannot rely on class name, there is another better way, define * options={"m2m_tags": True}* in field where define in kanban, not where we are showing, and add this.m2m_option as a dictionary and add those information in that, and change according to that rest of code.

**Comment by hmo-odoo** [comment]
can you write commit message base on guidelines: https://www.odoo.com/documentation/8.0/reference/guidelines.html#git

**Pull request closed without merging**